pkgs/tools/graphics/ldgallery/viewer/generate.sh

```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env nix-shell
+#!nix-shell -i bash -p nodePackages.node2nix
+
+# TODO: merge with other node packages in nixpkgs/pkgs/development/node-packages once
+# * support for npm projects in sub-directories is added to node2nix: https://github.com/svanderburg/node2nix/issues/177
+# * we find a way to enable development dependencies for some of the packages
+
+# Note for maintainers: keep in sync with the ldgallery compiler and sources in ./default.nix
+TAG="v2.0"
```

Comment:
You could extract the version from the package itself, like this: https://github.com/NixOS/nixpkgs/blob/master/pkgs/tools/X11/opentabletdriver/update.sh#L11

Reply:
I moved the TAG out as a parameter of the script and added some asserts to ensure that all versions are in sync.